ThingIFSDK/ThingIFSDK/QueryClause.swift

```diff
@@ -0,0 +1,368 @@
+        self.init(field, value: boolValue as AnyObject)
+    }
+
+    /** Get Equals clause for query as a Dictionary instance
+
+     - Returns: A Dictionary instance.
+     */
+    open override func makeDictionary() -> [ String : Any ] {
+        fatalError("TODO: implement me.")
+    }
+
+    /** Decoder confirming `NSCoding`. */
+    public required convenience init?(coder aDecoder: NSCoder) {
+        fatalError("TODO: implement me.")
+    }
+
+    /** Encoder confirming `NSCoding`. */
+    open override func encode(with aCoder: NSCoder) {
+        fatalError("TODO: implement me.")
+    }
+
+}
+
+/** Class represents Not Equals clause for query methods.  */
+open class NotEqualsClauseInQuery: QueryClause, BaseNotEquals {
+    public typealias EqualClauseType = EqualsClauseInQuery
+
+    /** Contained Equals clause instance. */
+    open let equals: EqualsClauseInQuery
+
+    public init(_ equals: EqualsClauseInQuery) {
+        self.equals = equals
+        super.init()
+    }
+
+    /** Get Not Equals clause for query as a Dictionary instance
+
+     - Returns: A Dictionary instance.
+     */
+    open override func makeDictionary() -> [ String : Any ] {
+        fatalError("TODO: implement me.")
+    }
+
+    /** Decoder confirming `NSCoding`. */
+    public required convenience init?(coder aDecoder: NSCoder) {
+        fatalError("TODO: implement me.")
+    }
+
+    /** Encoder confirming `NSCoding`. */
+    open override func encode(with aCoder: NSCoder) {
+        fatalError("TODO: implement me.")
+    }
+
+}
+
+/** Class represents Range clause for query methods. */
+open class RangeClauseInQuery: QueryClause, BaseRange {
+
+    /** Name of a field. */
+    open let field: String
+    /** lower limit for an instance. */
+    open let lower: (limit: AnyObject, included: Bool)?
+    /** upper limit for an instance. */
+    open let upper: (limit: AnyObject, included: Bool)?
+
+    private init(
+      _ field: String,
+      lower: (limit: AnyObject, included: Bool)? = nil,
+      upper: (limit: AnyObject, included: Bool)? = nil)
+    {
+        self.field = field
+        self.lower = lower
+        self.upper = upper
+        super.init()
+    }
+
+    /** Create Range clause for query having lower and upper limit.
+
+     - Parameter field: Name of the field to be compared.
+     - Parameter lower: lower limit
+       - limit: Limit value. The type must be integer, double or
+         float.
+       - included: Includes limit value or not. If true, limit value
+         is included. If false, limit value is not included
+     - Parameter upper: upper limit
+       - limit: Limit value. The type must be integer, double or
+         float.
+       - included: Includes limit value or not. If true, limit value
+         is included. If false, limit value is not included
+     - Returns: An instance of `RangeClauseInQuery`.
+     */
+    open static func range(
+      _ field: String,
+      lower: (limit: AnyObject, included: Bool),
```

Comment:
lowerLimit should not be `AnyObject`, better to limit it to number types(Int, Double, Float)